### PR TITLE
[Merged by Bors] - feat: port Computability.TMComputable

### PR DESCRIPTION
**Pull request by Komyyy**
---

- [x] depends on: #2799

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2799
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Num/Lemmas.lean [hunk 1]
```diff
@@ -0,0 +1,1820 @@
...
+  | pos a, pos b => PosNum.cast_add _ _
+  | pos a, neg b => by simpa only [sub_eq_add_neg] using PosNum.cast_sub' (α := α) _ _
+  | neg a, pos b =>
+    have : (↑b + -↑a : α) = -↑a + ↑b := by
+      rw [← PosNum.cast_to_int a, ← PosNum.cast_to_int b, ← Int.cast_neg, ← Int.cast_add (-a)]
+      simp [add_comm]
+    (PosNum.cast_sub' _ _).trans <| (sub_eq_add_neg _ _).trans this
+  | neg a, neg b =>
+    show -(↑(a + b) : α) = -a + -b by
+      rw [PosNum.cast_add, neg_eq_iff_neg_eq, neg_add_rev, neg_neg, neg_neg,
```
Comment (MichaelStollBayreuth) [inline Mathlib/Data/Num/Lemmas.lean:1335]: `neg_eq_iff_neq_eq` was removed by #17483. Use `neg_eq_iff_eq_neg` instead.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2799~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**